File: AutoJs6/src/core/Board.js
export default class Board {
  constructor(uic, pxProfile) {
    this.uic = uic;
    this.pxProfile = pxProfile;
    this.rect = this.pxProfile.BOARD_BOUNDARY_RECT;

    this.swipeSlotDuration = 0.005;
    this.swipePointOffset = 50;
  }

  convertToImageCoords(slot) {
    const [x, y] = slot;
    const [refX, refY] = this.pxProfile.BOARD_CENTER_REFERENCE_POINT;
    return [
      refX + this.pxProfile.SLOT_OFFSET_H * x,
      refY + this.pxProfile.SLOT_OFFSET_V * y,
    ];
  }

  convertToReducedImageCoords(slot) {
    const [x, y] = slot;
    return [
      this.pxProfile.REDUCED_SLOT_OFFSET_H * x,
      this.pxProfile.REDUCED_SLOT_OFFSET_V * y,
    ];
  }

  swipeSlot(slot1, slot2) {
    const [imgX1, imgY1] = this.convertToImageCoords(slot1);
    const [imgX2, imgY2] = this.convertToImageCoords(slot2);

    const [x1, y1] = slot1;
    const [x2, y2] = slot2;

    const offsetX = x1 === x2 ? 0 : this.swipePointOffset;
    const offsetY = y1 === y2 ? 0 : this.swipePointOffset;

    const signX = x1 < x2 ? 1 : -1;
    const signY = y1 < y2 ? 1 : -1;

    this.uic.swipe(
      imgX1 + offsetX * signX,
      imgY1 + offsetY * signY,
      imgX2 - offsetX * signX,
      imgY2 - offsetY * signY,
      this.swipeSlotDuration
    );
  }
}

File: PC/src/core/board.py
from .constants import *
from .dice_slot import DiceSlot
from .ui_controller import Uia2Controller



class Board:
    def __init__(self, uic: Uia2Controller, px_profile: PixelProfile):
        self.uic = uic
        self.px_profile = px_profile

        self.swipe_slot_duration = 0.005
        self.swipe_point_offset = 50

    def convert_to_image_coords(self, slot: DiceSlot):
        (x, y) = slot
        (ref_x, ref_y) = self.px_profile.BOARD_CENTER_REFERENCE_POINT
        return (
            ref_x + self.px_profile.SLOT_OFFSET_H*x,
            ref_y + self.px_profile.SLOT_OFFSET_V*y
        )

    def convert_to_reduced_image_coords(self, slot: DiceSlot) -> tuple[int, int]:
        (x, y) = slot
        return (
            self.px_profile.REDUCED_SLOT_OFFSET_H*x,
            self.px_profile.REDUCED_SLOT_OFFSET_V*y
        )

    def swipe_slot(self, slot1: DiceSlot, slot2: DiceSlot):
        img_x1, img_y1 = self.convert_to_image_coords(slot1)
        img_x2, img_y2 = self.convert_to_image_coords(slot2)
        (x1, y1) = slot1
        (x2, y2) = slot2
        offset_x = 0 if x1 == x2 else self.swipe_point_offset
        offset_y = 0 if y1 == y2 else self.swipe_point_offset
        sign_x = 1 if x1 < x2 else -1
        sign_y = 1 if y1 < y2 else -1
        self.uic.swipe(img_x1+offset_x*sign_x, img_y1+offset_y*sign_y,
                       img_x2-offset_x*sign_x, img_y2-offset_y*sign_y,
                       self.swipe_slot_duration)
